flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1231,22 +1257,26 @@ def run_to_convergence(self, desired_datasets, max_iterations=4):
             raise DidNotConverge(iteration_count=max_iterations)
 
     @given(
-        initial_dataset=DESIRED_DATASET_STRATEGY,
-        next_state=sampled_from([
-            DatasetStates.MOUNTED, DatasetStates.NON_MANIFEST,
-            DatasetStates.DELETED]),
+        two_dataset_states=TWO_DESIRED_DATASET_STRATEGY
     )
-    def test_simple_transitions(self, initial_dataset, next_state):
+    def test_simple_transitions(self, two_dataset_states):
         """
         Given an initial empty state, ``BlockDeviceCalculator`` will converge
         to any ``DesiredDataset``, followed by any other state of the same
         dataset.
         """
+        initial_dataset, next_dataset = two_dataset_states
+
         dataset_id = initial_dataset.dataset_id
-        initial_dataset = initial_dataset.set(
-            mount_point=self.deployer._mountpath_for_dataset_id(
-                unicode(dataset_id)),
-        )
+
+        # Set the mountpoint to a real mountpoint in desired dataset states
+        # that have a mount point attribute.
+        mount_point = self.deployer._mountpath_for_dataset_id(
+            unicode(dataset_id))
+        if hasattr(initial_dataset, 'mount_point'):
```

Comment:
:question: After the recent discussions about whether to use `hasattr` or `getattr`, is that something that we need to consider here?